Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -74,19 +74,21 @@ extension PackageGraph {
             // Sort all manifests toplogically.
             allManifests = try topologicalSort(inputManifests, successors: successors)
         }
-        var flattenedManifests: [String: GraphLoadingNode] = [:]
+
+        var flattenedManifests: [PackageIdentity: GraphLoadingNode] = [:]
         for node in allManifests {
-            if let existing = flattenedManifests[node.manifest.name] {
+            let packageIdentity = identityResolver.resolveIdentity(for: node.manifest.packageLocation)
+            if let existing = flattenedManifests[packageIdentity] {
                 let merged = GraphLoadingNode(
                     manifest: node.manifest,
                     productFilter: existing.productFilter.union(node.productFilter)
                 )
-                flattenedManifests[node.manifest.name] = merged
+                flattenedManifests[packageIdentity] = merged
             } else {
-                flattenedManifests[node.manifest.name] = node
+                flattenedManifests[packageIdentity] = node
             }
         }
-        allManifests = flattenedManifests.values.sorted(by: { $0.manifest.name < $1.manifest.name })
+        allManifests = flattenedManifests.values.sorted(by: { identityResolver.resolveIdentity(for: $0.manifest.packageLocation) < identityResolver.resolveIdentity(for: $1.manifest.packageLocation) })
```

Comment:
How expensive is the identity resolution (in theory, once it's complete, regardless of the current implementation)?  Should it be pulled out of the sort condition here?

Reply:
good point, eventually it will be cached by the resolver itself.